File: app/src/main/java/com/treelzebub/umap/api/discogs/model/Discogs.java
package com.treelzebub.umap.api.discogs.model;

import com.treelzebub.umap.api.AuthenticatedSession;

import java.util.ArrayList;
import java.util.List;

import retrofit.client.Response;

/**
 * Created by Tre Murillo 1/27/2015
 *
 * A class representing an authenticated Discogs.com session
 */

public class Discogs extends AuthenticatedSession {
    private static Discogs instance;
    public static Discogs getInstance() {
        return instance == null ? instance = new Discogs() : instance;
    }

}

File: app/src/main/kotlin/com/treelzebub/umap/api/discogs/Discogs.kt
package com.treelzebub.umap.api.discogs

import com.treelzebub.umap.api.discogs.model.User
import com.treelzebub.umap.api.discogs.model.Collection
import retrofit.client.Response
import retrofit.http.GET
import retrofit.http.Path

/**
 * Created by Tre Murillo on 6/6/15
 */

public interface Discogs {

    GET("/users/{user}")
    public fun getUser(Path("user") user: String): User

    GET("/users/{user}/collection/folders")
    public fun getCollection(Path("user") user: String): Collection
}